packages/augur-core/src/contracts/trading/erc20proxy1155/IERC20Proxy1155Nexus.sol

```diff
@@ -0,0 +1,40 @@
+pragma solidity 0.5.15;
+
+import 'ROOT/libraries/CloneFactory.sol';
+import 'ROOT/libraries/token/IERC20.sol';
+import 'ROOT/libraries/token/IERC1155.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+
+import 'ROOT/trading/erc20proxy1155/IERC20Proxy1155.sol';
+
+// To use these ERC20s, the user will have to call
+// setApprovalForAll(ERC20Proxy1155Nexus, true)
+// on the ShareToken contract themselves.
+
+contract IERC20Proxy1155Nexus is CloneFactory {
+    IERC1155 public target1155;
+
+    // tokenId -> owner -> spender -> allowance
+    // The spender is whoever called the proxy contract, which is msg.sender in the proxy and _caller here in the nexus.
+    mapping (uint256 => mapping (address => mapping (address => uint256))) public allowances;
```

Comment:
we're double defining this. Fine to put it on the interface contract IMO if its to help access via interface import

Reply:
of the two, proxies is useful elsewhere but allowances isn't. the allowance method provides that info.